.github/actions/validate-openapi-spec/action.yml

```diff
@@ -0,0 +1,30 @@
+name: validate-openapi-spec
+description: Validate OpenAPI specification compliance
+runs:
+  using: composite
+  steps:
+    - name: Install poetry
+      shell: bash
+      run: pipx install poetry
+
+    - name: Set up Python
+      uses: actions/setup-python@v5
+      with:
+        python-version: '3.11'
+        cache: 'poetry'
+
+    - name: Install package
+      shell: bash
+      run: poetry install -E all --only main
+
+    - name: Install openapi-spec-validator
+      shell: bash
+      run: poetry run pip install openapi-spec-validator
```

Comment:
Considering this is not a tool configured in the build system, I'd avoid using `poetry`, and simply use `pip` to execute.